llvm/cmake/modules/CheckProblematicConfigurations.cmake

```diff
@@ -11,7 +11,7 @@ endmacro()
 
 # MSVC and /arch:AVX is untested and have created problems before. See:
 # https://github.com/llvm/llvm-project/issues/54645
-if(${CMAKE_CXX_COMPILER_ID} STREQUAL MSVC)
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL MSVC)
```

Comment:
Based on the error message, it's likely that `CMAKE_CXX_COMPILER_ID` as a variable is empty or not defined. Expanding the variable here effectively turns this into `if(  STREQUAL MSVC)`, which is why the error complains about only seeing those two.

I would recommend changing your solution to:
`if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")`. This makes it clear that the former is a variable and the latter is a string.

Reply:
Sounds good! That seems to also be how we do it elsewhere in the codebase